packages/earljs/src/matchers/ObjectWith.ts

```diff
@@ -0,0 +1,37 @@
+import { smartEq } from '../validators/smartEq'
+import { Matcher } from './Base'
+
+/**
+ * Matches an object containing given key-value pairs.
+ */
+export class ObjectWithMatcher<T extends Object> extends Matcher {
+  constructor(private readonly expectedItem: T) {
+    super()
+  }
+
+  check(actualItem: unknown): boolean {
+    if (typeof actualItem !== 'object' || actualItem === null) {
+      return false
+    }
+
+    if ([Set.prototype, Map.prototype, Array.prototype].includes(Object.getPrototypeOf(actualItem))) {
+      return false
+    }
+
+    const expectedEntries = Object.entries(this.expectedItem)
+
+    return expectedEntries.every(([expectedKey, expectedValue]) => {
+      const actualValue = (actualItem as any)[expectedKey]
+
+      return smartEq(actualValue, expectedValue).result === 'success'
+    })
+  }
+
+  toString() {
+    return `[ObjectWith: ${this.expectedItem}]`
+  }
+
+  static make<T extends Object>(expectedItem: T): any {
```

Comment:
isn't type argument reduntant here? it's the same as
```
  static make(expectedItem: Object): any {
```

b/c of liskov substitution principle or something

Reply:
As we are not using the parametrized type in the return type, we can have it as an Object. Otherwise, I'd prefer to stick to the parametrized type as it'd be forwarded in the call chain.